From 5bfcdc4c8854e9b418d9402fd45533a8a2bd201e Mon Sep 17 00:00:00 2001
From: David Rubin <david@vortan.dev>
Date: Thu, 6 Nov 2025 20:12:35 -0800
Subject: [PATCH 046/133] llvm: clobber `rdx` instead of `edx` for x86-64
 valgrind request
X-Developer-Signature: v=1; a=openpgp-sha256; l=698; i=xtex@astrafall.org;
 h=from:subject; bh=MPCQ33Hg2IfefGvXG57R5Ot+IMtz/jXlV6iM851E7Yw=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78hheWLhduWfH136Ix+uLto9guMaJtP3zVkos3vDf6
 c+xlh8cHRNZGMS4GCzFFFmKDBu8WXXS+UWXlcvCzGFlAhkiLdLAwMDAwMLAl5uYV2qkY6Rnqm2o
 Z2ioY6RjxMDFKQBTbXmP4b8zn6Vf2dXLUXMCVxQlldXc/HTqYG6S3cvG0j9rKqLz7xky/BW9Ol/
 MUaflkKbvxM13vu3+kbNuYfHkL8pJz6Krue35FzADAA==
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

(cherry picked from commit 09e4035e79caea3174bb3f60ba005fe5838d82e5)
---
 src/codegen/llvm.zig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/codegen/llvm.zig b/src/codegen/llvm.zig
index fd0c08e23b98..1a9d1d869b2d 100644
--- a/src/codegen/llvm.zig
+++ b/src/codegen/llvm.zig
@@ -11805,7 +11805,7 @@ pub const FuncGen = struct {
                 \\ rolq  $$61, %rdi ; rolq $$51, %rdi
                 \\ xchgq %rbx, %rbx
                 ,
-                .constraints = "={rdx},{rax},{edx},~{cc},~{memory}",
+                .constraints = "={rdx},{rax},{rdx},~{cc},~{memory}",
             },
             else => unreachable,
         };
-- 
2.52.0

